widget/table.go

```diff
@@ -236,18 +339,41 @@ func (c *tableCells) Resize(s fyne.Size) {
 
 func (c *tableCells) Tapped(e *fyne.PointEvent) {
 	if e.Position.X < 0 || e.Position.X >= c.Size().Width || e.Position.Y < 0 || e.Position.Y >= c.Size().Height {
-		c.t.SelectedColumn = -1
-		c.t.SelectedRow = -1
+		c.t.selectedColumn = -1
+		c.t.selectedRow = -1
 		c.t.Refresh()
 		return
 	}
 
-	c.t.SelectedColumn = e.Position.X / (c.cellSize.Width + tableDividerThickness)
-	c.t.SelectedRow = e.Position.Y / (c.cellSize.Height + tableDividerThickness)
+	c.t.selectedColumn = e.Position.X / (c.cellSize.Width + tableDividerThickness)
+	c.t.selectedRow = e.Position.Y / (c.cellSize.Height + tableDividerThickness)
+
+	if c.t.OnSelectionChanged != nil {
+		c.t.OnSelectionChanged(c.t.selectedRow, c.t.selectedColumn)
+	}
+
+	if c.t.moveCallback != nil {
+		c.t.moveCallback()
+	}
```

Comment:
I wonder if the widget should be up-to-date before calling callbacks, ie call moveCallback before OnSelectionChanged